src/utils/nearest-wc-link-to-site.js

```diff
@@ -25,27 +42,46 @@ const getUserAssociatedWCLink = async (siteURI) => {
   const result = await fetch(url, { method: "GET", headers: headers });
 
   if (result.status === 404) {
-    return;
+    return null;
   } else {
     return await result.json();
   }
 };
 
+const setHighlightedWcLink = (map, wcLink) => {
+  const certaintyToColour = {
+    certain: "#06a516",
+    probable: "#ffec44",
+    ambiguous: "#9949cc",
+  };
+  map.setPaintProperty(
+    "highlightWatercourseLink",
+    "line-color",
+    certaintyToColour[wcLink.certainty] || "#06a516"
+  );
+  map.getSource("highlightWatercourseLink").setData(wcLink);
+};
+
 export const highlightNearestWatercourseLink = async (site, map) => {
   const siteURI = site.properties.uri;
   const userAssociatedWCLink = await getUserAssociatedWCLink(siteURI);
 
+  let allLinks = null;
   let wcLink = null;
+  let userSelected = false;
 
   if (userAssociatedWCLink?.status === "No watercourse link") {
     unhighlightWatercourseLink(map);
-    return;
+    return null;
   } else if (userAssociatedWCLink) {
     wcLink = userAssociatedWCLink;
+    userSelected = true;
```

Comment:
Do we need something like 
`allLinks.wcLinks = wcLink` here because I found that sites with a user selected wclink currently error cause on line 86, `allLinks` is still `null`

e.g. this one in Denham:
<img width="353" alt="image" src="https://user-images.githubusercontent.com/32230328/204004800-68607a40-f29b-47e7-adcb-d012c9c82d2b.png">

Reply:
good catch.. yeah we need those to still work. I went through several iterations trying to settle on some sort of data structure to return here. None are really ideal, but certainly need to work for all the cases. Thanks for catching this! I'll make sure it works with those use-selected links too!